lib/sheerid_api.rb

```diff
@@ -0,0 +1,65 @@
+# Service object for interacting with SheerID's API
+# docs found at: http://developer.sheerid.com/rest-api
+class SheeridAPI
+  DEFAULT_WEBHOOK_URL = 'https://dev.openstax.org/accounts/i/sheerid/webhook'
+  AUTHORIZATION_HEADER = "Bearer #{Rails.application.secrets.sheerid_api_secret}"
+  HEADERS = {
+    'Authorization': AUTHORIZATION_HEADER,
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  }.freeze
+
+  PROGRAM_ID = ENV['SHEERID_PROGRAM_ID']
+  BASE_URL = 'https://services.sheerid.com/rest/v2'
+
+  WEBHOOKS_URL = "#{BASE_URL}/program/#{PROGRAM_ID}/webhook"
+  INFO_URL = "#{BASE_URL}/info"
+  PROGRAM_THEME_URL = "#{BASE_URL}/#{PROGRAM_ID}/theme"
+
+  class << self
+    def get_verification_details(verification_id)
+      verification_id_details_url = "#{BASE_URL}/verification/#{verification_id}/details"
+      make_request(:get, verification_id_details_url)
+    end
+
+    private ###################
+
+    def make_request(http_method, url, body = nil)
+      begin
+        response = Faraday.send(http_method, url, body, HEADERS)
+        body = JSON.parse(response.body)
+        return body
+      rescue Net::ReadTimeout => ee
+        Raven.capture_message("SheeridAPI: timeout")
+        return nil
```

Comment:
returning nil is a bit of an anti-practice. better to return a object that has meaning (e.g, SheerApiResult).

Reply:
also, returns are needed here

Reply:
I gave it a shot last week and I reverted it because I didn't like it too much 🤷‍♂️ . One thing I wasn't sure about, for example, is where to define this "null object". I'm familiar with the Null Object pattern :).